libs/core/src/lib/select/select.component.ts

```diff
@@ -53,284 +87,288 @@ export interface OptionStatusChange {
 
+    /** Reference to root element for the mobile mode dialog */
+    @ViewChild('dialogContainer')
+    dialogContainerElementRef: ElementRef;
+
     /** Reference to element containing list of options */
     @ViewChild('selectOptionsListTemplate')
     selectOptionsListTemplate: TemplateRef<any>;
 
-    /** @hidden */
-    calculatedMaxHeight: number;
+    /** Custom template used to build control body. */
+    @Input()
+    controlTemplate: TemplateRef<any>;
 
-    /** Current selected option component reference. */
-    selected: OptionComponent;
 
-    /** Text value displayed in select control */
-    selectViewValue: string;
+    @Input()
+    mobile: boolean = false;
 
-    /** Whether popover is opened */
-    isOpen: boolean = false;
 
-    /** @hidden Cashed options as as Array */
-    private _options: OptionComponent[];
+    ////////////////////////////////////////////
+    /**
+     * Overlay pane containing the options.
+     *
+     * @hidden
+     */
+    @ViewChild(CdkConnectedOverlay)
+    overlayDir: CdkConnectedOverlay;
 
-    /** @hidden */
-    private _focusTrap: FocusTrap;
+    /**
+     * Referent to the option;s container
+     */
+    @ViewChild('optionPanel', { read: ElementRef })
+    optionPanel: ElementRef;
 
-    /** @hidden */
-    private _subscriptions: Subscription = new Subscription();
 
-    /** @hidden */
-    onChange: Function = () => {};
+    tabIndex: number;
 
     /** @hidden */
-    onTouched: Function = () => {};
+    onChange: Function = (value: any) => {
+    };
 
     /** @hidden */
-    @HostListener('keydown', ['$event'])
-    keydownHandler(event: KeyboardEvent): void {
-        if (KeyUtil.isKey(event, 'ArrowUp')) {
-            if (this.isInteractive) {
-                this._interactWithOptions('previous');
-            }
-            event.preventDefault();
-
-        } else if (KeyUtil.isKey(event, 'ArrowDown')) {
-            if (this.isInteractive) {
-                this._interactWithOptions('next');
-            }
-            event.preventDefault();
-
-        } else if (KeyUtil.isKey(event, 'Escape')) {
-            if (this.isInteractive) {
-                this.close();
-            }
-            event.preventDefault();
-
-        } else if (KeyUtil.isKey(event, [' ', 'Enter'])) {
-            if (this.isInteractive) {
-                this.toggle();
-            }
-            event.preventDefault();
-        }
-    }
+    onTouched: Function = () => {
+    };
 
-    /** @hidden */
-    @HostListener('window:resize')
-    resizeScrollHandler(): void {
-        this.calculatedMaxHeight = this.mobile ? null : window.innerHeight * 0.45;
-    }
 
     constructor(
-        private _elementRef: ElementRef,
-        private _selectProxy: SelectProxy,
-        private _changeDetectorRef: ChangeDetectorRef,
+        public _viewportRuler: ViewportRuler,
+        public  _elementRef: ElementRef,
+        public  _ngZone: NgZone,
+        public  _changeDetectorRef: ChangeDetectorRef,
```

Comment:
Do you add underscore only to implement common interface?

Reply:
This is the trouble with mixins. Again:
  - You want to make it accesible within the class
 - But you want other aware this is private API/Attribute dont use it. 

@JKMarkowski do you recomend other way to go around it? This is related only to those properties inside Mixins Interfaces.